src/components/stepforms/JoinStepForm.vue

```diff
@@ -105,10 +110,10 @@ export default class JoinStepForm extends BaseStepForm<JoinStep> {
     return this.availableDatasetNames.map(name => {
       const option = { label: name, trackBy: name };
       if (this.referencingPipelines.includes(name)) {
-        option['$isDisabled'] = true;
-        option[
-          'tooltip'
-        ] = `Circular reference: you cannot combine ${name} because it references the current dataset.`;
+        Object.assign(option, {
+          $isDisabled: true,
+          tooltip: `Circular reference: you cannot combine ${name} because it references the current dataset.`,
```

Comment:
Does that means we should add `$isDisabled` and `tooltip` as optional members of the `DropdownOption` interface?

Reply:
True!